src/lavinmq/mqtt/sparkplug/certificate_mapper.cr

```diff
@@ -0,0 +1,139 @@
+module LavinMQ
+  module MQTT
+    module Sparkplug
+      module CertificateMapper
+        CERTIFICATE_PREFIX = "$sparkplug/certificates/"
+
+        # Expands $sparkplug/certificates/# wildcard into actual BIRTH topic patterns
+        # Returns array of actual topic filters to subscribe to
+        #
+        # Examples:
+        #   "$sparkplug/certificates/#" -> ["spBv3.0/+/NBIRTH/+", "spBv3.0/+/DBIRTH/+/+"]
+        #   "$sparkplug/certificates/group1/#" -> ["spBv3.0/group1/NBIRTH/+", "spBv3.0/group1/DBIRTH/+/+"]
+        #   "$sparkplug/certificates/group1/NBIRTH/#" -> ["spBv3.0/group1/NBIRTH/+"]
+        #   "$sparkplug/certificates/group1/NBIRTH/node1" -> ["spBv3.0/group1/NBIRTH/node1"]
+        #   "$sparkplug/certificates/group1/DBIRTH/#" -> ["spBv3.0/group1/DBIRTH/+/+"]
+        #   "$sparkplug/certificates/group1/DBIRTH/node1/#" -> ["spBv3.0/group1/DBIRTH/node1/+"]
+        # ameba:disable Metrics/CyclomaticComplexity
+        def self.expand_certificate_subscription(filter : String) : Array(String)
+          # Remove certificate prefix
+          return [filter] unless filter.starts_with?(CERTIFICATE_PREFIX)
+
+          suffix = filter[CERTIFICATE_PREFIX.size..]
+          segments = suffix.split('/')
```

Comment:
It seems like we expect this to be at most 4 segments? I think this will do 2 allocations (well, one reallocation) for the array, so it's probably better to do
```suggestion
segments = Array(String).new(4)
suffix.split('/') { |segment| segments << segment }
```